File: cpp/src/fizzbuzz.cpp
#include "../headers/fizzbuzz.hpp"

FizzBuzz::~FizzBuzz() {}

string FizzBuzz::get(int number) {
    string result;
    addFizz(number, result);
    addBuzz(number, result);
    return result.empty() ? to_string(number) : result;
}

bool FizzBuzz::isFizz(int number) {
    return number % 3 == 0;
}

bool FizzBuzz::isBuzz(int number) {
    return number % 5 == 0;
}

void FizzBuzz::addFizz(int number, string& result) {
    if (isFizz(number)) {
        result += "Fizz";
    }
}

void FizzBuzz::addBuzz(int number, string& result) {
    if (isBuzz(number)) {
        result += "Buzz";
    }
}

File: ruby/lib/fizzbuzz.rb
# frozen_string_literal: true

module Fizzbuzz
  VERSION=2
  def Fizzbuzz.get(number)
    fizz = isFizz(number)
    buzz = isBuzz(number)
    if fizz and buzz
      return "FizzBuzz"
    elsif fizz
      return "Fizz"
    elsif buzz
      return "Buzz"
    end
    number.to_s()
  end

  def Fizzbuzz.isFizz(number)
    number % 3 == 0
  end

  def Fizzbuzz.isBuzz(number)
    number % 5 == 0
  end
end
